### PR TITLE
Fix #147912 (multipleSessionWarning debug option)

### LINKED ISSUE #147912: Add option to disable multiple debug sessions message
I want to run multiple debugging sessions at the same time, but when I start each next session, a warning dialog is shown.

Across post @mccambria in https://github.com/microsoft/vscode/issues/127721#issuecomment-917669506

### PR DESCRIPTION
**Pull request by dirondin**
Add new launch config debug option for silent `'{0}' is already running. Do you want to start another instance?` confirm dialog.

Tested with simple launch config:
```
{
    "command": "npm start",
    "name": "Run npm start",
    "request": "launch",
    "type": "node-terminal",
    "multipleSessionWarning": false
}
```

This PR fixes #147912

### CODE REVIEW

**Comment by dirondin** [comment]
Can anybody review this PR? :)

**Comment by roblourens** [comment]
Sorry for letting this slip through the cracks. I guess having this as a launch config option, rather than a user/workspace setting makes sense since there might be a particular launch config that you want to start multiple times?

I will take a look at this next week for July.

**Comment by roblourens** [comment]
Not sure about the checkbox, where we have options like that it usually stores it in some hidden state and does not write to a launch config

**Comment by dirondin** [comment]
Yes. The checkbox is an option for user/workspace or "hidden" location settings, not a launch configuration property. I just said that I would try to explore this option as well (in another PR).

**Comment by roblourens** [comment]
Feel free to explore it, but I'm not sure how I feel about that option, would take some thinking

File: src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts [hunk 1]
```diff
@@ -157,6 +157,11 @@ export class AdapterManager extends Disposable implements IAdapterManager {
 					},
 					'presentation': presentationSchema,
 					'internalConsoleOptions': INTERNAL_CONSOLE_OPTIONS_SCHEMA,
+					'multipleSessionWarning': {
+						type: 'boolean',
+						description: nls.localize('multipleSessionWarning', "Show a warning dialog when starting the same debug session."),
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts:162]: "another instance of the same debug session"

File: src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts [hunk 2]
```diff
@@ -157,6 +157,11 @@ export class AdapterManager extends Disposable implements IAdapterManager {
 					},
 					'presentation': presentationSchema,
 					'internalConsoleOptions': INTERNAL_CONSOLE_OPTIONS_SCHEMA,
+					'multipleSessionWarning': {
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts:160]: I'd prefer `suppressMultipleSessionWarning`

**Review by roblourens — COMMENTED** [review]
Thoughts @isidorn?

File: src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts (hunk 1 above)
Comment (weinand) [reply src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts:162]: A "debug session" is the result of starting a "debug configuration" (or starting "debug").
It is not possible to start "the same debug session" or "another instance of the same debug session".

My proposal:
"Warn when trying to start the same debug configuration more than once."

File: src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts (hunk 2 above)
Comment (weinand) [reply src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts:160]: `suppressMultipleSessionWarning` is OK, an alternative could be `suppressConcurrentSessionWarning`.

**Review by weinand — DISMISSED** [review]

**Comment by roblourens** [comment]
This looks good, but I'm out of the office and will merge it for August.

**Comment by isidorn** [comment]
@roblourens I am not a biggest fan of adding these "UI" settings to the launch.json. A better approach would probably be a setting. And yes I know that then it is not launch config specific, but I am worried if we keep it in a launch.json nobody will find it.

So a dialog + checkbox that writes to a setting is the way to go imho.
Rule of thumb - avoid UI attributes in launch configurations.

**Comment by roblourens** [comment]
To me, it's similar to the `presentation` section, which also sets metadata on how vscode should present/handle the launch config. I think that per-launch config makes the most sense, because starting multiple sessions of the same config is actually annoying, and I only want to avoid it for some specific configs.

**Review by roblourens — APPROVED** [review]

**Review by mjbvz — APPROVED** [review]

**Pull request merged**